jung-algorithms/src/main/java/edu/uci/ics/jung/layout/algorithms/FRBHVisitorLayoutAlgorithm.java

```diff
@@ -262,10 +262,8 @@ protected void calcRepulsion(N node1) {
     frNodeData.put(node1, Point.ORIGIN);
 
     ForceObject<N> nodeForceObject = new ForceObject<>(node1, layoutModel.apply(node1));
-
-    tree.visit(repulsion_constant, nodeForceObject);
-
-    frNodeData.put(node1, fvd1.add(nodeForceObject.getForce().x, nodeForceObject.getForce().y));
+    tree.visit(nodeForceObject, repulsion_constant);
+    frNodeData.put(node1, nodeForceObject.f);
   }
```

Comment:
this better duplicates what happens in the original FRLayout with the values for the node displacement.
The summation happens inside the visit and the end result is placed in the frNodeData